plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerClient.java

```diff
@@ -151,4 +164,63 @@ private void testImplementAggregation(AggregateFunction aggregateFunction, Map<S
             assertEquals(columnMapping.get().getType(), aggregateFunction.getOutputType());
         }
     }
+
+    @Test(dataProvider = "validTimestamps")
+    public void testVerifyValidTimestamps(int precision, LocalDateTime localDateTime, int picosOfMicro)
+    {
+        Instant instant = localDateTime.atZone(UTC).toInstant();
+        long epochMicros = instant.getEpochSecond() * MICROSECONDS_PER_SECOND + floorDiv(instant.getNano(), NANOSECONDS_PER_MICROSECOND);
+        picosOfMicro += (instant.getNano() % NANOSECONDS_PER_MICROSECOND) * PICOSECONDS_PER_NANOSECOND;
+        verifyTimestamp(epochMicros, picosOfMicro, precision);
+    }
+
+    @Test(dataProvider = "invalidTimestamps", expectedExceptions = VerifyException.class)
```

Comment:
please do not use `expectedExceptions`.
User `assertThatThrownBy(() -> {...}) instead.

Reply:
Done!